### PR TITLE
Core/Security:check for fake packets&bad addons that cause client crash

### PR DESCRIPTION
**Pull request by Viste**
##### CHANGES PROPOSED:

- add check for fake packets and disaster addons what make client crash

Closes: no issue for this 

to test  try crash client with some addon or fake packet 

##### Target branch(es):Master

### CODE REVIEW

File: src/server/game/Handlers/ChannelHandler.cpp [hunk 1]
```diff
@@ -56,6 +57,10 @@ void WorldSession::HandleLeaveChannel(WorldPacket& recvPacket)
     std::string channelName;
     recvPacket >> unk >> channelName;
 
+    //check for fake packets and bad addons that cause client to crash
+    if (!ChatHandler(this).isValidChatMessage(channelName.c_str()))
+        return;
```
Comment (BarbzYHOOL) [inline src/server/game/Handlers/ChannelHandler.cpp]: Why not make a small function for this? to avoid duplicate code

Comment (Viste) [reply src/server/game/Handlers/ChannelHandler.cpp]: lazzy and its old good code )

Comment (Viste) [reply src/server/game/Handlers/ChannelHandler.cpp]: oh ok :)

**Comment by Viste** [comment]
ChatFake and etc

**Comment by Viste** [comment]
https://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-exploits/376771-exploit-make-other-players-crash.html

**Comment by Viste** [comment]
ant other

**Comment by BarbzYHOOL** [comment]
Awesome PR though. Do you have a commit reference (TC, Mangos) ?

**Comment by Viste** [comment]
> Awesome PR though. Do you have a commit reference (TC, Mangos) ?
no its old patch from trinity-core.ru :)

**Comment by BarbzYHOOL** [comment]
Ok, even better

**Comment by wetbrownsauce** [comment]
I can't seem to replicate the crash with any of those ownedcore links so I'm not sure how to test :S

**Comment by BarbzYHOOL** [comment]
Isn't it related to this worldserver conf `ChatFakeMessagePreventing = 1`

**Comment by Viste** [comment]
@wetbrownsauce i don't remember name of all addons(only ChatFake) what have all client crash funcs >_< just google first what can find )

File: src/server/game/Handlers/ChannelHandler.cpp (hunk 1 above)
Comment (FrancescoBorzi) [reply src/server/game/Handlers/ChannelHandler.cpp]: @Viste can you please update this PR with this suggestion?

Comment (Viste) [reply src/server/game/Handlers/ChannelHandler.cpp]: yes after on next week

**Pull request closed without merging**